File: Java/Arrays&Strings/isPalindrome.java
class isPalindrome {
    // Method to check if the given string is a palindrome
    public boolean isPalindrome(String s) {
        String processedString = "";

        for (int i = 0; i < s.length(); i++) {
            if (isAlphanumeric(s.charAt(i))) {
                processedString += Character.toLowerCase(s.charAt(i));
            }
        }

        int left = 0;
        int right = processedString.length() - 1;

        while (left < right) {
            if (processedString.charAt(left) != processedString.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }

    boolean isAlphanumeric(char character) {
        return Character.isLetterOrDigit(character);
    }

    public static void main(String[] args) {
        isPalindrome solution = new isPalindrome();

        System.out.println(solution.isPalindrome("abcdcba"));
        System.out.println(solution.isPalindrome("racecar"));
        System.out.println(solution.isPalindrome("A man, a plan, a canal: Panama"));
        System.out.println(solution.isPalindrome("race a car"));
        System.out.println(solution.isPalindrome(" "));
    }
}

File: Javascript/Arrays&Strings/isPalindrome.js
function isPalindrome(s) {
    let processedString = "";

    for (let i = 0; i < s.length; i++) {
        if (isAlphanumeric(s[i])) {
            processedString += s[i].toLowerCase();
        }
    }

    let left = 0;
    let right = processedString.length - 1;

    while (left < right) {
        if (processedString[left] !== processedString[right]) {
            return false;
        }
        left++;
        right--;
    }

    return true;

    function isAlphanumeric(char) {
        return /^[a-z0-9]+$/i.test(char);
    }
}



console.log(isPalindrome("abcdcba"));
console.log(isPalindrome("racecar"));
console.log(isPalindrome("A man, a plan, a canal: Panama"));
console.log(isPalindrome("race a car"));
console.log(isPalindrome(" "));